File: tkpdfeed/src/main/java/com/tokopedia/tkpd/tkpdfeed/feedplus/domain/model/topads/ProductImage.java
package com.tokopedia.tkpd.tkpdfeed.feedplus.domain.model.topads;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author by errysuprayogi on 4/3/17.
 * Copied to feed by milhamj 1/18/17.
 */
public class ProductImage {
    private String m_url;
    private String s_url;
    private String xs_url;
    private String m_ecs;
    private String s_ecs;
    private String xs_ecs;

    public ProductImage(String m_url, String s_url, String xs_url, String m_ecs, String s_ecs,
                        String xs_ecs) {
        this.m_url = m_url;
        this.s_url = s_url;
        this.xs_url = xs_url;
        this.m_ecs = m_ecs;
        this.s_ecs = s_ecs;
        this.xs_ecs = xs_ecs;
    }

    public String getM_url() {
        return m_url;
    }

    public void setM_url(String m_url) {
        this.m_url = m_url;
    }

    public String getS_url() {
        return s_url;
    }

    public void setS_url(String s_url) {
        this.s_url = s_url;
    }

    public String getXs_url() {
        return xs_url;
    }

    public void setXs_url(String xs_url) {
        this.xs_url = xs_url;
    }

    public String getM_ecs() {
        return m_ecs;
    }

    public void setM_ecs(String m_ecs) {
        this.m_ecs = m_ecs;
    }

    public String getS_ecs() {
        return s_ecs;
    }

    public void setS_ecs(String s_ecs) {
        this.s_ecs = s_ecs;
    }

    public String getXs_ecs() {
        return xs_ecs;
    }

    public void setXs_ecs(String xs_ecs) {
        this.xs_ecs = xs_ecs;
    }
}

File: features/discovery/topadssdk/src/main/java/com/tokopedia/topads/sdk/domain/model/ProductImage.kt
package com.tokopedia.topads.sdk.domain.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.tokopedia.kotlin.model.ImpressHolder
import kotlinx.android.parcel.Parcelize
import org.json.JSONObject

private const val KEY_M_URL = "m_url"
private const val KEY_S_URL = "s_url"
private const val KEY_XS_URL = "xs_url"
private const val KEY_M_ECS = "m_ecs"
private const val KEY_S_ECS = "s_ecs"
private const val KEY_XS_ECS = "xs_ecs"

@Parcelize
data class ProductImage(
    @SerializedName(KEY_M_URL)
    var m_url: String = "",

    @SerializedName(KEY_S_URL)
    var s_url: String = "",

    @SerializedName(KEY_XS_URL)
    var xs_url: String = "",

    @SerializedName(KEY_M_ECS)
    var m_ecs: String = "",

    @SerializedName(KEY_S_ECS)
    var s_ecs: String = "",

    @SerializedName(KEY_XS_ECS)
    var xs_ecs: String = ""
) : ImpressHolder(), Parcelable
